### ISSUE TITLE
Profiles heatmap

### ISSUE THREAD

**Issue by kolesnikovae** [grafana/pyroscope#3015]
I propose the addition of a profiles heatmap alongside the current time-series visualization for profiling data.

<img width="1652" alt="image" src="https://github.com/grafana/pyroscope/assets/12090599/d9570289-8fc9-47ae-ab7d-2c3d7eb9a1f7">

(_this is how it looked in Pyroscope cloud_)

In this representation:

Each cell corresponds to a group of profiles meeting the XY criteria:
 - X: Time range
 - Y: Sum of sample values per profile (illustrated as CPU seconds in the example)

Z axis is encoded in color and indicates the number of profiles in the cell (normalised)

Users should be able to select specific areas on the heatmap and visualize the profiling data for the chosen cell or region. Additionally, the feature should support the comparison or differentiation of selected areas.

**Comment by fandreuz** [comment]
Hi @kolesnikovae, is there any ECD for this feature? Thanks!

**Comment by kolesnikovae** [comment]
Hi @fandreuz, although we don't have any estimates yet, I can say that we've completed all the core changes needed to make this possible. Some of my colleagues are already working on a similar feature – profile exemplars: https://github.com/grafana/pyroscope/pull/4615 – and I hope heatmaps will follow soon after.

cc @simonswine

**Comment by simonswine** [comment]
Yes you should see some progress there soon, we first like to finish the exemplars work, but stay tuned.
